glfw31-gl41core-triangle/triangle.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2014 The go-gl Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Renders a simple static triangle using GLFW 3.1 and OpenGL 4.1 core forward-compatible profile.
+package main
+
+import (
+	"fmt"
+	"runtime"
+
+	gl "github.com/go-gl/gl/v4.1-core/gl"
+	glfw "github.com/go-gl/glfw/v3.1/glfw"
+)
+
+const windowWidth = 800
+const windowHeight = 640
+
+var vertexShader = `
+#version 410
+
+in vec3 pos;
+out vec3 position; // used in fragment shader
+
+void main() {
+  position = pos;
+  gl_Position = vec4(pos, 1.0);
+}
+` + "\x00"
+
+var fragmentShader = `
+#version 410
+
+in vec3 position; // comes from vertex shader
+out vec4 outColor;
+
+void main() {
+  outColor = vec4(position+0.5, 1.0); // adds 0.5 to make it pretty (shifts the colors)
+}
+` + "\x00"
+
+func main() {
+	runtime.LockOSThread()
```

Comment:
This is bad. `LockOSThread` is not guaranteed to lock the OS thread unless it's in a package init, really. While this probably works today, there is no guarantee it will work in the future.

Place it inside an init like here: https://github.com/go-gl/examples/blob/master/glfw31-gl41core-cube/cube.go#L26-L29